clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -176,6 +181,15 @@ void runOnce() {
      */
     private void processApplicationEvents() {
         LinkedList<ApplicationEvent> events = new LinkedList<>();
+
+        if (!networkClientDelegate.hasAnyPendingRequests()) {
+            // If we don't have any outstanding network requests, we can park here until the next event comes.
+            ApplicationEvent head = applicationEventQueue.poll();
```

Comment:
Are you sure about this? My understanding is that the thread won't be able to process any inbound packets/bytes if they arrive before a new event comes in here to unblock it.